### PR TITLE
Technical architecture overview

### LINKED ISSUE #1030: Add an overview of the architecture of the site.
Expand beyond what is summarized in the [Folder Structure](https://github.com/Qiskit/qiskit.org/#-folder-structure) section of the README.md to explain how the site is architected including:
- Components.
- Integrations with external services.
- Markdown content rendering (old and new styles).
- Implementation of Carbon Design System

### PR DESCRIPTION
**Pull request by eddybrando**
Expand technical architecture overview section in the **README.md** file as suggested in #1030.

The newly created **ARCHITECTURE.md** aims to serve as a simple but coherent and complete overview of how the website is architected, without going into too much detail.

It should be seen as a general guide of our current practices and a good onboarding document for project newcomers.

Some updates have been also made to the **README.md** file to reflect the current directory structure and fix minor typos.

---

Closes #1030

### CODE REVIEW

**Review by techtolentino — APPROVED** [review]
Such a good addition to the project documentation. Thank you @eddybrando for the thoroughness.

**Review by delapuente — CHANGES_REQUESTED** [review]
The general comment is that the content is too shallow. We should make it deeper and try to add more value to the developer.

Also, it would be convenient to tell how some parts connect with others.

**Comment by delapuente** [comment]
How is this goind @eddybrando ? Do you have any blocker?

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+## Vue.js components
+
+We write our components as [single file components](https://vuejs.org/v2/guide/single-file-components.html).
+
+The **script section** is written in [class-style syntax](https://class-component.vuejs.org/) with TypeScript and uses decorators provided by [**vue-property-decorator**](https://github.com/kaorun343/vue-property-decorator).
+
+The **style section** is written in SCSS.
+
+Here is an example of a component called `GreetingMessage.vue`:
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply ARCHITECTURE.md:23]: Followed up on https://github.com/Qiskit/qiskit.org/pull/1654#discussion_r607798635 and addressed on #1654

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+## Markdown content
+
+We use Markdown files to add content and have two different approaches for this:
+
+- content (old)
+- new-content (new)
+
+These approaches currently coexist but we aim to replace the former with the latter.
+
+### content (old)
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply ARCHITECTURE.md:81]: Followed up on https://github.com/Qiskit/qiskit.org/pull/1654/files#r607821692 and addressed with #1654

**Comment by delapuente** [comment]
Superseded by #1654

**Pull request closed without merging**